File: backend/app/schemas/compromisso.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from uuid import UUID

class CompromissoBase(BaseModel):
    titulo: str = Field(..., min_length=3, max_length=255)
    descricao: Optional[str] = None
    tipo: str = Field(..., description="Tipo do compromisso: Visita, Instalação, Medição, etc")
    status: str = Field(default="Agendado")
    data_hora_inicio: datetime
    data_hora_fim: datetime
    local: Optional[str] = None
    endereco: Optional[str] = None
    observacoes: Optional[str] = None
    cliente_id: Optional[UUID] = None
    pedido_id: Optional[UUID] = None

class CompromissoCreate(CompromissoBase):
    pass

class CompromissoUpdate(BaseModel):
    titulo: Optional[str] = Field(None, min_length=3, max_length=255)
    descricao: Optional[str] = None
    tipo: Optional[str] = None
    status: Optional[str] = None
    data_hora_inicio: Optional[datetime] = None
    data_hora_fim: Optional[datetime] = None
    local: Optional[str] = None
    endereco: Optional[str] = None
    observacoes: Optional[str] = None
    cliente_id: Optional[UUID] = None
    pedido_id: Optional[UUID] = None

class CompromissoRead(CompromissoBase):
    id: UUID
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/compromisso.ts
export interface CompromissoBase {
  titulo: string;
  descricao?: string;
  tipo: string; // 'Visita', 'Instalação', 'Medição'
  status: string; // 'Agendado', 'Concluído', 'Cancelado'
  data_hora_inicio: string; // ISO string
  data_hora_fim: string; // ISO string
  local?: string;
  endereco?: string;
  observacoes?: string;
  cliente_id?: string;
  pedido_id?: string;
}

export interface CompromissoCreate extends CompromissoBase {}

export interface CompromissoUpdate extends Partial<CompromissoBase> {}

export interface CompromissoRead extends CompromissoBase {
  id: string;
  tenant_id: string;
  created_at: string;
  updated_at: string;
  // Futuramente podemos adicionar objetos aninhados de Cliente e Pedido aqui se o backend retornar
}